torch/autograd/gradcheck.py

```diff
@@ -289,7 +289,7 @@ def _as_tuple(x):
 
 
 def _differentiable_outputs(x):
-    return tuple(o for o in _as_tuple(x) if o.requires_grad)
+    return tuple(o for o in _as_tuple(x) if torch.is_tensor(o) and o.requires_grad)
```

Comment:
Why do you need to change gradcheck here?
I feel like this should be in a separate PR because that may have some unexpected impact as well....

Reply:
I added gradcheck as per @soulitzer's suggestion. I guess it makes sense to fix `gradcheck` as well as part of this PR since without this, someone could call `gradcheck` on the output of a custom autograd function and it would fail.

Reply:
Right but is this exercised / tested?

Reply:
Yes, I've added a test for this. See last line of `test_custom_function_non_tensor_inputs_outputs`.